ironmon_tracker/Program.lua

```diff
@@ -157,6 +176,35 @@ function Program.stepFrames()
 	Program.Frames.carouselActive = Program.Frames.carouselActive + 1
 end
 
+function Program.updateRepelSteps()
+	-- Checks for an active repel and updates the current steps remaining
+	-- Game uses a variable for the repel steps remaining, which remains at 0 when there's no active repel
+	local saveblock1Addr = Utils.getSaveBlock1Addr()
+	local repelStepCountOffset = Utils.inlineIf(GameSettings.game == 3, 0x40, 0x42)
+	local repelStepCount = Memory.readword(saveblock1Addr + GameSettings.gameVarsOffset + repelStepCountOffset)
+	if repelStepCount ~= nil and repelStepCount > 0 then
+		Program.ActiveRepel.inUse = true
+		if repelStepCount ~= Program.ActiveRepel.stepCount then
+			Program.ActiveRepel.stepCount = repelStepCount
+			-- Duration is defaulted to normal repel (100 steps), check if super or max is used instead
```

Comment:
Nope, what this does is initially sets the duration to 100 (so standard repel) and if the read-in remaining step count is larger than that it will change the duration to either 200 or 250 depending on the step count.

In the example of a super repel:
1) Super repel gets used
2) The variable being read from memory contains a value of 200 (the steps remaining)
3) Tracker reads this, 200 > 100 therefore it will check for what to change the duration to. As the value is 200, it will set the duration to 200 instead of 100. I used a less than or equals to in case they have e.g. 172/200 steps left and refresh the tracker or something.
4) The repel icon displayed will have a bar showing progress for 0-200 steps remaining

The main downside of this is if the setting is disabled/re-enabled or the tracker stopped and restarted while a repel is active, the duration resets to the default value of 100 and it'll just take the current remaining steps value for this duration check. So if less than 100 steps remaining on whatever repel (normal, super, max) it'll just show a progress bar from 0-100 like a regular repel (which is still kinda fine imo?)

I considered just checking the special vars item id address (the one we use for the fishing rod) but figured i'd do it this way to have one less memory read, plus i don't really know how the QoL patches that let you use a repel again when one runs out without going into your bag again interact with that memory address so didn't wanna risk it not getting the right reads there

Reply:
Gotcha, this makes sense to me. Thank you for the explanation.